clipper_admin/clipper_admin/deployers/onnx.py

```diff
@@ -0,0 +1,173 @@
+    It serializes the PyTorch model in Onnx format and creates a container that loads it as a Caffe2 model.
+    Parameters
+    ----------
+    clipper_conn : :py:meth:`clipper_admin.ClipperConnection`
+        A ``ClipperConnection`` object connected to a running Clipper cluster.
+    name : str
+        The name to be assigned to both the registered application and deployed model.
+    version : str
+        The version to assign this model. Versions must be unique on a per-model
+        basis, but may be re-used across different models.
+    input_type : str
+        The input_type to be associated with the registered app and deployed model.
+        One of "integers", "floats", "doubles", "bytes", or "strings".
+    inputs :
+        input of func.
+    func : function
+        The prediction function. Any state associated with the function will be
+        captured via closure capture and pickled with Cloudpickle.
+    pytorch_model : pytorch model object
+        The Pytorch model to save.
+    base_image : str, optional
+        The base Docker image to build the new model image from. This
+        image should contain all code necessary to run a Clipper model
+        container RPC client.
+    labels : list(str), optional
+        A list of strings annotating the model. These are ignored by Clipper
+        and used purely for user annotations.
+    registry : str, optional
+        The Docker container registry to push the freshly built model to. Note
+        that if you are running Clipper on Kubernetes, this registry must be accesible
+        to the Kubernetes cluster in order to fetch the container from the registry.
+    num_replicas : int, optional
+        The number of replicas of the model to create. The number of replicas
+        for a model can be changed at any time with
+        :py:meth:`clipper.ClipperConnection.set_num_replicas`.
+    onnx_backend : str, optional
+        The provided onnx backend.Caffe2 is the only currently supported ONNX backend.
+    """
+    if base_image is None:
+        if onnx_backend is "caffe2":
+            base_image = "clipper/caffe2-onnx-container:{}".format(__version__)
+        else:
+            logger.error(
+                "{backend} ONNX backend is not currently supported.".format(
+                    backend=onnx_backend))
+
+    serialization_dir = save_python_function(name, func)
+
+    try:
+        torch_out = torch.onnx._export(
+            pytorch_model, inputs, "model.onnx", export_params=True)
+
+    except Exception as e:
+        logger.error(
+            "Error serializing PyTorch model to ONNX: {e}".format(e=e))
+
+    logger.info("Torch model has be serialized to ONNX format")
+
+    # Deploy model
+    clipper_conn.build_and_deploy_model(name, version, input_type,
+                                        serialization_dir, base_image, labels,
+                                        registry, num_replicas, batch_size)
```

Comment:
Move the call to `clipper_conn.build_and_deploy_model()` to inside the try statement